addons/redis/scripts/redis-sentinel-start-v2.sh

```diff
@@ -42,6 +42,124 @@ build_redis_sentinel_conf() {
+        output=""
+        for sentinel_pod_fqdn in "${sentinel_pod_fqdn_list[@]}"; do
+          if [ -n "$SENTINEL_PASSWORD" ]; then
+            temp_output=$(redis-cli -h "$sentinel_pod_fqdn" -p 26379 -a "$SENTINEL_PASSWORD" sentinel masters 2>/dev/null || true)
+          else
+            temp_output=$(redis-cli -h "$sentinel_pod_fqdn" -p 26379 sentinel masters 2>/dev/null || true)
+          fi
+          if [[ -n "$temp_output" ]]; then
+              output="$temp_output"
+              break
+          fi
+        done
+    else
+        echo "SENTINEL_POD_FQDN_LIST environment variable is not set or empty."
+        return 1
+    fi
+    #TODO:Check if the sentinel has been deleted before, as adding a new sentinel might read the past configuration, leading to conflicts with the current setup.
+    if [[ -n "$output" ]]; then
+        master_name=""
+        master_ip=""
+        master_port=""
+        master_down_after_milliseconds=""
+        master_quorum=""
+        master_failover_timeout=""
+        master_parallel_syncs=""
+        while read -r line; do
+            case "$line" in
+                name)
+                    read -r master_name
+                    ;;
+                ip)
+                    read -r master_ip
+                    ;;
+                port)
+                    read -r master_port
+                    ;;
+                down-after-milliseconds)
+                    read -r master_down_after_milliseconds
+                    ;;
+                quorum)
+                    read -r master_quorum
+                    ;;
+                failover-timeout)
+                    read -r master_failover_timeout
+                    ;;
+                parallel-syncs)
+                    read -r master_parallel_syncs
+                    ;;
+            esac
+
+            if [[ -n "$master_name" && -n "$master_ip" && -n "$master_port" && \
+                  -n "$master_down_after_milliseconds" && -n "$master_failover_timeout" && \
+                  -n "$master_parallel_syncs" && -n "$master_quorum" ]]; then
+            echo "Master Name: $master_name, IP: $master_ip, Port: $master_port, \
+            down-after-milliseconds: $master_down_after_milliseconds, \
+            failover-timeout: $master_failover_timeout, \
+            parallel-syncs: $master_parallel_syncs, quorum: $master_quorum"
+
+            echo "sentinel monitor $master_name $master_ip $master_port $master_quorum" >> /data/sentinel/redis-sentinel.conf
+            echo "sentinel down-after-milliseconds $master_name $master_down_after_milliseconds" >> /data/sentinel/redis-sentinel.conf
+            echo "sentinel failover-timeout $master_name $master_failover_timeout" >> /data/sentinel/redis-sentinel.conf
+            echo "sentinel parallel-syncs $master_name $master_parallel_syncs" >> /data/sentinel/redis-sentinel.conf
+            echo "sentinel auth-user $master_name $REDIS_SENTINEL_USER" >> /data/sentinel/redis-sentinel.conf
```

Comment:
If the master name is configured with duplicates in the Sentinel configuration file, it will cause the Sentinel instances to fail to start.